dht_net.go

```diff
@@ -108,41 +112,60 @@ func (dht *IpfsDHT) handleNewMessage(s inet.Stream) {
 	}
 }
 
+func tagWithMessage(ctx context.Context, name string, msg *pb.Message) {
+	log.SetTag(ctx, name+".key", b58.Encode([]byte(msg.GetKey())))
+	log.SetTag(ctx, name+".type", msg.GetType().String())
+}
+
 // sendRequest sends out a request, but also makes sure to
 // measure the RTT for latency measurements.
 func (dht *IpfsDHT) sendRequest(ctx context.Context, p peer.ID, pmes *pb.Message) (*pb.Message, error) {
+	ctx = log.Start(ctx, "sendRequest")
+	defer log.Finish(ctx)
+	log.SetTag(ctx, "peer.address", p.Pretty())
+	log.SetTag(ctx, "self.address", dht.self.Pretty())
+	tagWithMessage(ctx, "request", pmes)
 
 	ms, err := dht.messageSenderForPeer(p)
 	if err != nil {
+		log.SetErr(ctx, err)
 		return nil, err
 	}
 
 	start := time.Now()
 
 	rpmes, err := ms.SendRequest(ctx, pmes)
 	if err != nil {
+		log.SetErr(ctx, err)
 		return nil, err
 	}
+	tagWithMessage(ctx, "response", rpmes)
```

Comment:
I don't see the benefit of tagging the response and request differently. The key and type should be identical.

Reply:
The reason for this is the request/response are part of the same span, so if we tag them the same, one will overwrite the other. We could convert this to span events, but that'll be ugly as well since span events only support single k/v (no nesting) AFAIK.